okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -344,6 +347,73 @@ static Status toGrpcStatus(ErrorCode code) {
     return ERROR_CODE_TO_STATUS.get(code);
   }
 
+  private Socket upgradeToTls(Socket socket, SSLSocketFactory sslSocketFactory) throws IOException {
+    SSLSocket sslSocket = (SSLSocket) sslSocketFactory.createSocket(
+        socket, authorityHost, address.getPort(), true /* auto close */);
+    final String[] supportedTlsProtocols = sslSocket.getSupportedProtocols();
+    Set<String> supportedTlsProtocolsSet = new HashSet<String>(supportedTlsProtocols.length);
+    for (int i = 0; i < supportedTlsProtocols.length; ++i) {
+      supportedTlsProtocolsSet.add(supportedTlsProtocols[i]);
+    }
+    List<String> enableTlsProtocols = new ArrayList<String>();
+    addIfSupported(supportedTlsProtocolsSet, enableTlsProtocols,
+        "TLSv1.2", "TLSv1.1", "TLSv1");
+    sslSocket.setEnabledProtocols(
+        enableTlsProtocols.toArray(new String[enableTlsProtocols.size()]));
+
+    final String[] supportedCiphers = sslSocket.getSupportedCipherSuites();
+    Set<String> supportedCiphersSet = new HashSet<String>(supportedCiphers.length);
+    for (int i = 0; i < supportedCiphers.length; ++i) {
```

Comment:
Actually, just do:
```java
Set<String> supportedCiphersSet = new HashSet<String>(Arrays.asList(supportedCiphers));
```

It will auto-size the set.